[Dev docs] [update-content] [1477635] (#2739)
MicrosoftDocs/msteams-docs: msteams-platform/concepts/deploy-and-publish/apps-upload.md

title: Upload your custom app
description: Describes how to upload your app in Microsoft Teams
ms.topic: how-to

> ms.author: 

Reviewer: Do not keep the field empty. Add La's name here. lajanuar

Some of these apps are intended to work in a team or a group chat, with personal app experience being secondary.
Hence, the default install scope selection helps you to specify the `defaultInstallScope` for the apps that you publish. The app installation experience makes the default options available to the user, while the rest is moved under the chevron as highlighted in the image.

![Add an app](../../assets/images/compose-extensions/addanapp.png)

The `defaultInstallScope` property supports values, such as personal, team, group chat, or meeting.

> [!NOTE]
> For a successful validation, specify a `defaultInstallScope` that supports all capabilities. Each capability in an app must support the `defaultInstallScope`, else the app acts as it would without the default installation scope.

`defaultGroupCapability` defines capabilities that are installed by default. Choose tab, bot, or connector as the default capability of the app for each group scope. The app validation fails if any of the scopes do not support the default capability.

## Remove or update your app

> To remove your app, select the delete icon next to the app name in the View Teams bots list.

Reviewer: Check the formatting.

Author: corrected

> If you change manifest information, you must first remove the app and then add the updated package per [Load your package into a team](#load-your-package-into-teams). Code changes on your service do not require you to re-upload your manifest, unless those changes require manifest updates, such as changes to the URL or the Microsoft app ID for its bot.

Reviewer: Check if we can use **re-upload** as per MSTP. 
```suggestion
If you change manifest information, first remove the app and then add the updated package, see [Load your package into a team](#load-your-package-into-teams). Code changes on your service do not require you to upload your manifest again. However, if the code changes require manifest updates, such as changes to the URL or the Microsoft app ID for its bot, you must upload the manifest again.
```

> [!NOTE]
> You cannot remove a bot from a personal context entirely. If the bot is removed and added again, additional communication with the bot appends to the previous conversation.

## Troubleshooting notes

> If the manifest fails to load, check that you have followed all the instructions in [Create the package](../../concepts/build-and-test/apps-package.md) and validated your manifest against the [schema](../../resources/schema/manifest-schema.md).

Reviewer: ```suggestion
If the manifest fails to load, check if you have followed all the instructions in [Create the package](../../concepts/build-and-test/apps-package.md) and validated your manifest against the [schema](../../resources/schema/manifest-schema.md).
```

  ![The Add a tab dialog box, featuring a gallery of available tabs.](../../assets/images/connector_gallery.png)

## Access your uploaded messaging extension

An uploaded app with a messaging extension automatically appears in the *More options* (*&#8943;*) menu in the compose box.

![Messaging extensions](../../assets/images/compose-extensions/cesampleapp.png)

## Add an app

Although installing an app in the personal scope works for most apps, some of the apps in Teams Store support both personal and team scopes.
Some of these apps are intended to work in a team or a group chat, with personal app experience being secondary.
Hence, the default install scope selection helps you to specify the `defaultInstallScope` for the apps that you publish. The app installation experience makes the default options available to the user, while the rest is moved under the chevron as highlighted in the image.

![Add an app](../../assets/images/compose-extensions/addanapp.png)

> In the specified app manifest, the property `defaultInstallScope` supports values like personal, team, group chat, or meeting.

Reviewer: What is 'specified' app manifest? Where is is 'specified'?

> [!NOTE]

> > Ensure your app specifies a `defaultInstallScope` that is supported by all the capabilities in the app for the validation to pass. In other words, every single capability in an app must support the `defaultInstallScope`. Otherwise, the behavior reverts to the experience where no default installation scope is provided.

Reviewer: Reword and make it precise. The intent of the note is not coming out.
